### PR TITLE
chore: add e2e for in-app messaging and backfill test for existing notifications

### PR DESCRIPTION
**Pull request by awsluja**
#### Description of changes
This adds test coverage for pre-existing notification channels (SMS), and ensure compatibility between those channels and the new channel (In-App Messaging). It also make sure that both SMS/In-App Messaging channels continue to work well with Pinpoint/Analytics.

These test cases ensure the following flows work, and that configuration files are set correctly.

1. adding/removing notifications in a new project (SMS)
2. adding/removing notifications in a new project (in-app messaging)
3. adding/removing notifications in a project that has pinpoint configured already (SMS)
4. adding/removing notifications in a project that has pinpoint configured already (in-app messaging)

Additionally, several checks for file configuration states, locally & in the cloud, are included.

See slack for information on test cases.

E2E is running at: https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/11743/workflows/cae05854-8bab-4983-bc75-f77b4a910640

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-e2e-tests/src/__tests__/notifications-analytics-compatibility-sms-1.test.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    // SETUP NOTIFICATIONS CHANNEL (SMS, pushes inline)
+    const settings = { resourceName: pinpointResourceName };
+    await addNotificationChannel(projectRoot, settings, testChannelSelection, true, true);
+
+    await sleep(3000);
+    // SMS & Analytics meta should exist
+    const meta = getBackendAmplifyMeta(projectRoot);
+    console.log(meta.analytics);
+    console.log(meta.notifications[pinpointResourceName]?.output);
+    const SMSMeta = meta.notifications[pinpointResourceName]?.output?.SMS;
```
Comment (lazpavel) [inline packages/amplify-e2e-tests/src/__tests__/notifications-analytics-compatibility-sms-1.test.ts:71]: nit: `smsMeta`

File: packages/amplify-e2e-tests/src/__tests__/notifications-analytics-compatibility-sms-1.test.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    const analyticsMeta = meta.analytics[pinpointResourceName]?.output;
+    expect(SMSMeta).toBeDefined();
+    expect(analyticsMeta).toBeDefined();
+    expect(SMSMeta.Enabled).toBe(true);
+    expect(SMSMeta.ApplicationId).toEqual(analyticsMeta.Id);
+
+    // pinpointId in team-provider-info should match the analyticsMetaId
+    const teamInfo = getTeamProviderInfo(projectRoot);
+    console.log(teamInfo[envName].categories);
+    // const pinpointId = teamInfo[envName].categories?.notifications?.Pinpoint?.Id;
```
Comment (lazpavel) [inline packages/amplify-e2e-tests/src/__tests__/notifications-analytics-compatibility-sms-1.test.ts:81]: should this comment be removed?

File: packages/amplify-e2e-tests/src/__tests__/notifications-analytics-compatibility-in-app-2.test.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+    // BEGIN - SETUP PINPOINT & PUSH (see analytics.test.ts)
+    await addPinpointAnalytics(projectRoot, pinpointResourceName);
+    await amplifyPushUpdate(projectRoot);
+
+    // SETUP NOTIFICATIONS CHANNEL & PUSH (IN-APP MESSAGING)
+    const settings = { resourceName: pinpointResourceName };
+    await addNotificationChannel(projectRoot, settings, 'In-App Messaging', true, true);
+
+    // PUTH NOTIFICATIONS
```
Comment (lazpavel) [inline packages/amplify-e2e-tests/src/__tests__/notifications-analytics-compatibility-in-app-2.test.ts:62]: push

**Review by lazpavel — APPROVED** [review]
small nits, otherwise LGTM, will approve when conflicts are resolved

File: packages/amplify-e2e-tests/src/__tests__/notifications-analytics-compatibility-in-app-1.test.ts [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    const pinpointResourceName = `${projectPrefix}${getShortId()}`;
+    await initJSProjectWithProfile(projectRoot, projectSettings);
+
+    const appId = getAppId(projectRoot);
+    expect(appId).toBeDefined();
+
+    // BEGIN - SETUP PINPOINT BUT DON'T PUSH (see analytics.test.ts)
+    await addPinpointAnalytics(projectRoot, pinpointResourceName);
+
+    // SETUP NOTIFICATIONS CHANNEL BUT DON'T PUSH (IN-APP MESSAGING)
```
Comment (lazpavel) [inline packages/amplify-e2e-tests/src/__tests__/notifications-analytics-compatibility-in-app-1.test.ts:62]: super nit: not a fan of caps comments

Comment (sobolk) [reply packages/amplify-e2e-tests/src/__tests__/notifications-analytics-compatibility-in-app-1.test.ts:62]: +1. additionally not a fan of comments. can the code be structured in such a way (or is it already) that one can read control flow from function names ?

File: packages/amplify-category-notifications/src/commands/notifications/remove.ts [hunk 5]
```diff
@@ -68,13 +68,14 @@ export const run = async (context: $TSContext): Promise<$TSContext> => {
...
         await writeData(context, channelAPIResponse);
+        printer.info('The channel has been successfully updated.');
       }
     } else if (isPinpointAppOwnedByNotifications(pinpointAppStatus.status)) {
       const confirmDelete = await prompter.confirmContinue('Confirm that you want to delete the associated Amazon Pinpoint application');
       if (confirmDelete) {
         await deletePinpointApp(context);
-        printer.info('The Pinpoint application has been successfully deleted.');
         await writeData(context, undefined);
+        printer.info('The Pinpoint application has been successfully deleted.');
```
Comment (sobolk) [inline packages/amplify-category-notifications/src/commands/notifications/remove.ts:78]: why did we remove `successfully` from channel-sms but we have it here ?

**Review by sobolk — DISMISSED** [review]

**Comment by awsluja** [comment]
closing due to weird rebase state, going to re-open as a new PR

**Pull request closed without merging**